src/runtime/contrib/clml/clml_runtime.cc

```diff
@@ -144,6 +144,8 @@ class CLMLRuntime : public JSONRuntimeBase {
     workspace = cl::OpenCLWorkspace::Global();
     workspace->Init();
     tentry = workspace->GetThreadEntry();
+    device_id = workspace->GetCLDeviceID(tentry->device.device_id);
+    platform_id = workspace->device_to_platform[device_id];
```

Comment:
If there is no OpenCL devices presented, then an exception will be generated here. Is it ok to you? Or I suggest moving these two lines after if condition `if (!ExtensionStringPresent())`.

Reply:
It's unlikely to have multiple OpenCL devices or no OpenCL device on Snapdragon for now.

Anyway, I will keep the initialization flow clean by looking for ```ExtensionStringPresent``` followed by other assignments.